luigi/scheduler.py

```diff
@@ -642,7 +663,16 @@ def _remove_workers_from_tasks(self, workers, remove_stakeholders=True):
     def disable_workers(self, worker_ids):
         self._remove_workers_from_tasks(worker_ids, remove_stakeholders=False)
         for worker_id in worker_ids:
-            self.get_worker(worker_id).disabled = True
+            worker = self.get_worker(worker_id)
+            worker.disabled = True
+            self._update_metrics_worker_status_change(worker, 'disabled')
+
+    def _update_metrics_task_status_change(self, task, status):
+        if status != UNKNOWN:
```

Comment:
Not sure this is the correct place for this check, but tasks can get added to the state before we actually know anything about them. I wish there was a better way to identify when new tasks are being added.